### PR TITLE
🌱 E2E: Cleanup emulator variable

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**:

We are now specifying the BMC protocol directly and do not need to expose the emulator variable. The emulator will be picked based on the protocol.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: hack/ci-e2e.sh [hunk 1]
```diff
@@ -3,12 +3,11 @@
...
 #              BMO project. It uses either vbmc or sushy-tools based on
-#              the BMO_E2E_EMULATOR environment variable.
-#              With sushy-tools, it is also possible to choose between
-#              redfish-virtualmedia and redfish protocols using the
-#              SUSHY_TOOLS_PROTOCOL environment variable.
-#              By default, sushy-tools and redfish-virtualmedia will be used.
-# Usage:       export BMO_E2E_EMULATOR="vbmc"  # Or "sushy-tools"
+#              the BMC_PROTOCOL environment variable.
+#              Supported protocols are: ipmi, redfish and redfish-virtualmedia.
+#              VBMC is used for ipmi and sushy-tools for the both redfish protocols.
```
Comment (elfosardo) [inline hack/ci-e2e.sh:8]: nit: "for both redfish protocols"

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Review by elfosardo — COMMENTED** [review]
/lgtm

File: hack/ci-e2e.sh (hunk 1 above)
Comment (lentzi90) [reply hack/ci-e2e.sh]: Thanks, fixed!

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Review by kashifest — COMMENTED** [review]
/approve
Much easier to understand now. Thanks for the cleanup

**Pull request merged**